extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/ConfigBuildStep.java

```diff
@@ -268,8 +268,13 @@ void generateConfigProperties(
             BuildProducer<ConfigClassBuildItem> configClasses,
             BuildProducer<AdditionalConstrainedClassBuildItem> additionalConstrainedClasses) {
 
-        processConfigClasses(combinedIndex, generatedClasses, reflectiveClasses, reflectiveMethods, configClasses,
+        Map<String, GeneratedClassBuildItem> generatedConfigClasses = new HashMap<>();
```

Comment:
(Not very familiar with the code-generation code, so ignore my comment, if it doesn't apply)

Should this be a `LinkedHashMap` for reproducibility?